tests/test_pipedrive_bookmarks.py

```diff
@@ -0,0 +1,162 @@
+        # setting 'second_start_date' as bookmark for running 2nd sync
+        for stream, updated_state in simulated_states.items():
+            new_state['bookmarks'][stream] = updated_state
+
+        # Set state for next sync
+        menagerie.set_state(conn_id, new_state)
+
+        ##########################################################################
+        ### Second Sync
+        ##########################################################################
+
+        # Run a sync job using orchestrator
+        second_sync_record_count = self.run_and_verify_sync(conn_id)
+        second_sync_records = runner.get_records_from_target_output()
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        for stream in expected_streams:            
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_replication_method = expected_replication_methods[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in
+                                       first_sync_records.get(
+                                           stream, {}).get('messages', [])
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(
+                                            stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = list(expected_replication_keys[stream])[0] #Key in which state has been saved in state file
+                    record_replication_key = list(expected_replication_keys[stream])[0]
+
+                    first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                    second_bookmark_value = second_bookmark_key_value.get(replication_key)
+                    
+                    first_bookmark_value_ts = self.dt_to_ts(first_bookmark_value)
+                    
+                    second_bookmark_value_ts = self.dt_to_ts(second_bookmark_value)
+                    
+                    simulated_bookmark_value = self.dt_to_ts(new_state['bookmarks'][stream][replication_key])
```

Comment:
This should be hardcoding the format that it is using to parse to datetime, so that there is not the potential to miss a bug if that format changes unexpectedly.